Sources/Drill/Pass.swift

```diff
@@ -0,0 +1,144 @@
+/// Pass.swift
+///
+/// Copyright 2017, The Silt Language Project.
+///
+/// This project is released under the MIT license, a copy of which is
+/// available in the repository.
+
+import Foundation
+import Lithosphere
+
+/// A class that's passed to invocations of each pass. It contains a timer
+/// and diagnostic engine that each pass can make use of.
+class PassContext {
```

Comment:
Do you anticipate the passes having separate contexts?  If not then this can be ` final`.

Reply:
Good catch!